cmd/podman/machine/platform_windows.go

```diff
@@ -1,19 +1,40 @@
 package machine
 
 import (
+	"fmt"
 	"os"
 
+	"github.com/containers/common/pkg/config"
 	"github.com/containers/podman/v4/pkg/machine"
 	"github.com/containers/podman/v4/pkg/machine/hyperv"
 	"github.com/containers/podman/v4/pkg/machine/wsl"
 )
 
-func GetSystemDefaultProvider() machine.VirtProvider {
-	// This is a work-around for default provider on windows while
-	// hyperv is one developer.
-	// TODO this needs to be changed back
-	if _, exists := os.LookupEnv("HYPERV"); exists {
-		return hyperv.GetVirtualizationProvider()
+func GetSystemProvider(vmType *machine.VMType) (machine.VirtProvider, error) {
```

Comment:
What is the reason for the argument? Is there a case where we need it? or worst case, write it unexported with a exported caller?

Reply:
Open extension point to support override passed via command line parameter. This is what I had in mind and why it is an exported func. I don't know if there will be a need/desire to have command line parameter, though.

Can remove it, if you see that simplification is more important and the use case is too artificial.